### PR TITLE
Add peering endpoint API docs

### PR DESCRIPTION
**Pull request by freddygv**
### Description
Documentation for the endpoints to create, read, list, and delete peerings.

### CODE REVIEW

File: website/content/api-docs/peering.mdx [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+The table below shows this endpoint's support for
+[blocking queries](/api-docs/features/blocking),
+[consistency modes](/api-docs/features/consistency),
+[agent caching](/api-docs/features/caching), and
+[required ACLs](/api#authentication).
+
+| Blocking Queries | Consistency Modes | Agent Caching | ACL Required  |
+| ---------------- | ----------------- | ------------- | ------------- |
+| `YES`            | `consistent`      | `none`        | `none`        |
```
Comment (kisunji) [inline website/content/api-docs/peering.mdx:282]:
```suggestion
| `NO`            | `consistent`      | `none`        | `none`        |
```

File: website/content/api-docs/peering.mdx [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+page_title: Cluster Peering - HTTP API
+description: The /peering endpoints allow for managing cluster peerings.
+---
+
+# Cluster Peering - HTTP API
+
+~> This page covers features that are currently in _technical preview_. Features and
+functionality are subject to change. You should never use the technical preview release
+in secure environments or production scenarios. Features in technical preview may face
+performance and scaling issues, with limited support options available.
```
Comment (im2nguyen) [inline website/content/api-docs/peering.mdx:12]:
```suggestion
~> **Cluster peering is currently in technical preview:** Functionality associated with cluster peering is subject to change. You should never use the technical preview release in secure environments or production scenarios. Features in technical preview may have performance issues, scaling issues, and limited support.
```

Comment (im2nguyen) [reply website/content/api-docs/peering.mdx:12]: this is what we have for the normal cluster peering docs

**Review by im2nguyen — APPROVED** [review]
Overall, it looks good from a style stand-point. Can you change the destination branch to `docs-cluster-peering-technical-preview` so that we can merge both set of cluster peering docs in teh same PR? 

Thank you!

File: website/content/api-docs/peering.mdx [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+
+- `Datacenter` `(string: "")` - Specifies the datacenter, which defaults to the
+  agent's datacenter if not provided.
+
+- `Token` `(string: "")` - Specifies the ACL token to use in the request instead of
+  the one specified via the `token` query parameter, `X-Consul-Token` request header,
+  or `CONSUL_HTTP_TOKEN` environment variable.
+
+- `Meta` `(map<string|string>: <optional>)` - Specifies arbitrary KV metadata
+  to associate with the peering.
```
[the remark this answers is not in the document]
Comment (freddygv) [reply website/content/api-docs/peering.mdx]: These metadata pairs are effectively labels that may have meaning to the user but not to Consul.

I added the sentence you suggested.

File: website/content/api-docs/peering.mdx [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+This endpoint marks a peering for deletion. Once marked, Consul will
+delete all the data imported from the peer in the background. Only once
+all associated data has been deleted will the peering actually disappear.
+Until then, further reads can be performed on the peering and a `DeletedAt`
+field will now be populated with the timestamp of when the peering was
+marked for deletion.
+
+| Method   | Path               | Produces |
+| -------- | ------------------ | -------- |
+| `DELETE` | `/peering/:name`   | N/A      |
```
[the remark this answers is not in the document]
Comment (freddygv) [reply website/content/api-docs/peering.mdx:219]: Yep, the colon indicates that the name here is provided by the user

**Pull request merged**